docs/glossary.md

```diff
@@ -0,0 +1,25 @@
+(_proxy_py_glossary)=
+# {{ project }} Glossary
+
+```{spelling}
+DNS
+DoH
+Py
+```
+
+```{glossary}
+DoH
+DNS over HTTP
+DNS-over-HTTP
+
+  [DNS over HTTP]
+```
+
+```{glossary}
+DNS
```

Comment:
@abhinavsingh these are declared mostly so that they could be used via a `term` directive instead of adding them to spelling exclusions. It's also useful for interlinking the same concept from various places. The lists may contain a few aliases as demonstrated for DoH. RST usage example:
```rst
:term:`DNS`
:term:`a custom DNS mention anchor <DNS>`
```

Reply:
All of this is awesomeness.  I did notice the glossary section :)